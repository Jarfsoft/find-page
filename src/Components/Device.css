.device-card {
  width: 85%;
  margin: 0 auto;
  margin-bottom: 2rem;
  background-color: white
}

.device-card img {
  width: 100%;
}

.info {
  display: flex;
  justify-content: space-between;
  margin: 0 1rem;
  font-size: 0.8rem;
  padding: 1.5rem 0;
}

a,
span {
  text-decoration: none;
  color: gray;
}
