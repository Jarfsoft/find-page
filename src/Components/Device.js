/* eslint-disable react/forbid-prop-types */

import { React } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './Device.css';

function Device({ data }) {
  return (
    <div className="device-card">
      <img alt="Device" src={data.image_url} />
      <div className="info">
        <Link to={`/device/${data.id}`}>{data.name}</Link>
        <span>{`$ ${data.price}`}</span>
      </div>
    </div>
  );
}

Device.propTypes = {
  data: PropTypes.object.isRequired,
};

export default Device;

/* eslint-enable react/forbid-prop-types */
