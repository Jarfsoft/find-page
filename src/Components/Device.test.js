import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import Device from './Device';

describe('Navbar render', () => {
  const mockStore = configureStore();
  let store;
  const state = {
    devices: [],
  };

  describe('Snapshot check', () => {
    it('should render Device component', () => {
      store = mockStore(state);
      const device = renderer
        .create(
          <Provider store={store}>
            <BrowserRouter>
              <Device
                imageUrl="image"
                id="id"
                name="name"
                price="price"
              />
            </BrowserRouter>
          </Provider>,
        ).toJSON();
      expect(device).toMatchSnapshot();
    });
  });
  it('should containt an image', () => {
    store = mockStore(state);
    const device = render(
      <BrowserRouter>
        <Provider store={store}>
          <Device
            imageUrl="image"
            id="id"
            name="name"
            price="price"
          />
        </Provider>
      </BrowserRouter>,
    );
    const img = device.container.querySelectorAll('img');
    expect(img).toHaveLength(1);
  });
});
