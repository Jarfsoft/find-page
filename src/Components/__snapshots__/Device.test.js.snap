// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render Snapshot check should render Device component 1`] = `
<div
  className="device-card"
>
  <img
    alt="Device"
    src="image"
  />
  <div
    className="info"
  >
    <a
      href="/device/id"
      onClick={[Function]}
    >
      name
    </a>
    <span>
      $ price
    </span>
  </div>
</div>
`;
